torch/distributed/_sharded_tensor/api.py

```diff
@@ -540,7 +543,12 @@ def sharding_spec(self) -> ShardingSpec:
         return self._sharding_spec
 
     def __torch_function__(self, func, types, args=(), kwargs=None):
-        raise RuntimeError(f"torch function '{func.__name__}' not supported for ShardedTensor!")
+        if func == torch.nn.functional.linear:
```

Comment:
so this dispatch happens for 

sharded_linear(input) --> nn.functional.linear(input, weight, bias, ...) 

b/c weight is ShardedTensor ?

Reply:
nice!